### DISCUSSION TITLE
Can I run a Python file in C++ function with PythonQt

### DISCUSSION THREAD

**Discussion by gaosh97** [MeVisLab/pythonqt#213] [Q&A] [answered]
Can I run python file in c++ function with PythonQt, my python file like this
[test.txt](https://github.com/MeVisLab/pythonqt/files/15062140/test.txt)

**Comment by usiems** [comment, marked as answer]
Sorry I didn't answer earlier: You _can_ run a Python file with PythonQt, but you don't need PythonQt for that.
You can initialize a normal Python interpreter in your code and tell it to compile and execute your code. PythonQt doesn't make it that much easier to do that. What PythonQt is good at is making your own C++ objects available in Python. What it is very good at is making QObjects (the standard interface object of the Qt framework) available in Python.
